packages/design-system/src/components/Drawer/DrawerToggle.tsx

```diff
@@ -0,0 +1,68 @@
+import Button from '../Button/Button';
+import React, { useEffect, useRef } from 'react';
+import classNames from 'classnames';
+
+export interface DrawerToggleProps {
+  /**
+   * Determines if Drawer is open or closed.
+   * This value is used to re-focus the toggle that opened the drawer when the drawer closes.
+   */
+  drawerOpen: boolean;
+  /**
+   * DrawerToggle content.
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classes for the toggle button anchor element.
+   */
+  className?: string;
+  /**
+   * Adds `display: inline` to the DrawerToggle.
+   */
+  inline?: boolean;
+  /**
+   * This function is called with an id that the toggle generates.
+   * It can be used in implementing the Drawer for keeping track of which drawer the toggle controls.
+   */
+  showDrawer: (string) => any;
+}
+
+/**
+ * A link that triggers the visibility of a drawer
+ */
+const DrawerToggle = ({
+  className,
+  children,
+  inline,
+  showDrawer,
+  drawerOpen,
+  ...others
+}: DrawerToggleProps) => {
+  const buttonRef = useRef(null);
+
+  useEffect(() => {
+    if (!drawerOpen && buttonRef) {
```

Comment:
I think `buttonRef.current` is what you want to check, because the `buttonRef` should always exist because it's created on line 41.